### PR TITLE
WEBAPP-375: Generalize event detail and page

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://integreat.atlassian.net/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
This PR makes the `Page` redundant (at least for the WebApp), but since it is exported and probably also used (in the Wohnraumboerse frontend), I can't remove it. How should I deal with it?

File: src/modules/endpoint/models/DateModel.js [hunk 1]
```diff
@@ -24,6 +24,37 @@ class DateModel {
   get allDay (): boolean {
     return this._allDay
   }
+
+  /**
+   * Returns a formatted string containing all relevant start and end date and time information
+   * @param {String} locale The locale to format the span in
+   * @return {String} The formatted span string
+   */
+  toFormattedString (locale: string): string {
```
Comment (klinzo) [inline src/modules/endpoint/models/DateModel.js]: @maxammann I would leave this in the model if it's okay. I like it better like this and think it makes things a little bit easier.

File: src/modules/common/components/Page.js [hunk 2]
```diff
@@ -1,24 +1,48 @@
...
+  object-fit: contain;
+`
 
 type PropsType = {|
   title: string,
+  thumbnail?: string,
   content: string,
-  onInternLinkClick: string => void,
-  hijackRegExp?: RegExp
+  lastUpdate: Moment,
```
Comment (Schedulaar) [inline src/modules/common/components/Page.js]: Please make lastUpdate optional - we might not want to show a lastUpdate on every Page (in the wohnraumbörse, we use the Page component and we probably don't want to show a last-update there)

Comment (klinzo) [reply src/modules/common/components/Page.js]: Done

File: src/modules/endpoint/models/DateModel.js (hunk 1 above)
Comment (maxammann) [reply src/modules/endpoint/models/DateModel.js]: Imo this is code to render a two timestamp and belongs next to a render method, but you can leave it here.

Comment (klinzo) [reply src/modules/endpoint/models/DateModel.js]: @Schedulaar what do you think?

Comment (Schedulaar) [reply src/modules/endpoint/models/DateModel.js]: I don't really mind 😅

**Pull request merged**